File: TryOuts/apiCallBackend.py
from http.server import BaseHTTPRequestHandler, HTTPServer
import json
import coreapi


class APIRequestHandler(BaseHTTPRequestHandler):

    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')  # Enable CORS for all origins
        self.end_headers()

        with open('loginData.txt') as f:
            data = f.readlines()
            username = data[0].strip()
            password = data[1].strip()

        client = coreapi.Client()
        schema = client.get("http://86.119.43.87/docs/")
        action = ["account", "create"]
        params = {
            "username": username,
            "password": password,
        }
        result = client.action(schema, action, params=params)
        response = json.dumps(result)
        self.wfile.write(bytes(response, "utf8"))


def run_server():
    server_address = ('localhost', 3000)
    httpd = HTTPServer(server_address, APIRequestHandler)
    print('Starting server on port 3000...')
    httpd.serve_forever()


if __name__ == '__main__':
    run_server()

File: TryOuts/apiCallBackend.js
const http = require('http');
const fs = require('fs');
const coreapi = require('coreapi');
const NodeTransport = require('coreapi-node-transport');

const server = http.createServer((req, res) => {
    if (req.method === 'GET') {
        res.writeHead(200, {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*'
        });

        fs.readFile('loginData.txt', 'utf8', (err, data) => {
            if (err) throw err;

            const [username, password] = data.trim().split('\n');

            const transport = new NodeTransport('http://86.119.43.87/docs/');
            const client = new coreapi.Client({ transport: transport });
            client.action(['account', 'create'], { username: username, password: password }).then((result) => {
                res.end(JSON.stringify(result));
            }).catch((err) => {
                console.error(err);
                res.statusCode = 500;
                res.end('Internal Server Error');
            });
        });
    } else {
        res.statusCode = 405;
        res.end('Method Not Allowed');
    }
});

server.listen(3000, () => {
    console.log('Server running at http://localhost:3000/');
});
